pyrealm/core/calendar.py

```diff
@@ -118,3 +118,78 @@ def __repr__(self) -> str:
         """Representation of a Calendar instance."""
 
         return f"Calendar({self.dates[0]!s}, {self.dates[-1]!s})"
+
+
+@dataclass
+class LocationDateTime:
+    """A data class representing an observation location and date and time information.
+
+    This class encapsulates the latitude and longitude of a location along with a
+    corresponding time array. It automatically calculates the latitude and longitude in
+    radians, the Julian days from the date-time information, and a decimal
+    representation of the local time.
+
+    Example:
+        >>> import numpy as np
+        >>> ldt = LocationDateTime(
+        ...     latitude=-35.058333,
+        ...     longitude=147.34167,
+        ...     UTC_offset=10,
+        ...     year_date_time=np.array([np.datetime64("2024-08-12T10:30")])
+        ... )
+        >>> print(ldt.latitude_rad)
+        -0.6118833411105811
+        >>> print(ldt.local_time)
+        [10.5]
+    """
+
+    latitude: float
+    """The latitude of the location in degrees."""
+    latitude_rad: float = field(init=False)
+    """The latitude of the location in radians, calculated automatically."""
+    longitude: float
+    """The longitude of the location in degrees."""
+    longitude_rad: float = field(init=False)
+    """The longitude of the location in radians, calculated automatically."""
+    UTC_offset: int
+    """The offset from Coordinated Universal Time (UTC) for the location."""
+    year_date_time: np.ndarray
+    """An array of np.datetime64 values corresponding to observations at the 
+    location."""
+    julian_days: np.ndarray = field(init=False)
+    """An array of Julian day of the year numbers calculated from the
+    ``year_date_time``."""
+    local_time: np.ndarray = field(init=False)
+    """An array of Julian day of the year numbers calculated from the 
+    ``year_date_time``."""
```

Comment:
The docstring is wrong here 😄 

Also, it might be worth changing `LocationDateTime.local_time`  to `LocationDateTime.decimal_time` or `LocationDateTime.local_decimal_time` to make it clear that its a different representation.

Reply:
Done